gdal/frmts/georaster/oci_wrapper.cpp

```diff
@@ -1583,6 +1595,38 @@ char* OWStatement::ReadCLob( OCILobLocator* phLocator )
     return pszBuffer;
 }
 
+void OWStatement::FreeTempLob(OCILobLocator* phLocator)
+{
+    boolean        is_temporary;
+
+    if (phLocator == nullptr)
+        return;
+
+    if( CheckError( OCILobIsTemporary(
+        poConnection->hEnv,
+        hError,
+        phLocator,
+        &is_temporary), 
+        hError))
+    {
+        CPLDebug("OCI", "OCILobIsTemporary failed");
+        return;
+    }
+
+    if(is_temporary)
+    {
+      if( CheckError( OCILobFreeTemporary(
+        poConnection->hSvcCtx,
+        hError,
+        phLocator),
+        hError))
+      {
+        CPLDebug("OCI", "OCILobFreeTemporary failed");
+        return;
+      }
+    }
```

Comment:
Would be nice to add a call to OCIDescriptorFree() here so you don't call it from the outside. 

Not a deal breaker.

Reply:
It has benefits on either way. Moving OCIDescriptorFree() inside FreeTempLob() makes code cleaner, however it is less explicit from the caller level whether the OCI descriptor has been freed or not.

Reply:
It seems like the bug was based on the wrong assumption that OCIDescriptionFree would free lob locators. If that is the case, I think you should also look into the method OWStatement::Free() where all the array of RDT lob locators are freed. It might have the same issue and a big impact overall.

Moving the OCIDescriptionFree to the inside of FreeTempLob() should be safe since the only use for that API is currently georaster_wrapper.cpp (and PDAL maybe) and as I mentioned in the previous paragraph, the intention was to free the lob locator.

Reply:
I did look at OWStatement::Free(). If I understand correctly, it is for releasing the LOB locator descriptor for the LOBs in the RDT table, which are permanent LOBs. There is no need to free the permanent LOBs. So I don't think we need to free the LOBs there unless it is handling the temporary LOB. 

I will move the OCIDescriptorFree to the inside of FreeTempLob(). Do I need to create a new pull request or I can work on the same pull request and how?

Reply:
You should proceed with this PR but probably going to need to **squash** into a single commit.

But it says here you need "write permission" to do that: https://help.github.com/articles/about-pull-request-merges/

Let's see what Even suggests.

Reply:
The article on github is for maintainers that merge PR. Here it is just a matter of ammending the commit with "git commit -a --amend" and force pushing it over with "git push -f your_github_nick your_branch_name"

Reply:
The updated code has been committed and pushed to the repository.  I also would like to add the fix to 2.2 and 2.3 branch, what is the process? Do I need to wait for this pull request to be closed?